catalyst_voices/lib/widgets/indicators/voices_no_internet_connection_banner.dart

```diff
@@ -0,0 +1,90 @@
+  final VoidCallback onRefresh;
+
+  const NoInternetConnectionBanner({
+    super.key,
+    this.onRefresh = _noop,
+  });
+
+  static void _noop() {}
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        final shouldButtonDisplay = constraints.maxWidth > 744;
+
+        return Container(
+          padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
+          margin: const EdgeInsets.symmetric(horizontal: 16),
+          decoration: BoxDecoration(
+            color: VoicesColors.darkAvatarsError,
+            borderRadius: BorderRadius.circular(8),
+          ),
+          child: Row(
+            mainAxisSize: MainAxisSize.min,
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Expanded(
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Row(
+                      mainAxisSize: MainAxisSize.min,
+                      children: [
+                        CatalystSvgIcon.asset(
+                          VoicesAssets.icons.wifi.path,
+                          color: VoicesColors.darkIconsOnImage,
+                        ),
+                        const SizedBox(width: 8),
+                        Expanded(
+                          child: Text(
+                            context.l10n.noConnectionBannerTitle,
+                            style: Theme.of(context)
+                                .textTheme
+                                .titleLarge!
+                                .copyWith(
+                                  color: VoicesColors.darkTextOnPrimaryLevel0,
+                                ),
+                          ),
+                        ),
+                      ],
+                    ),
+                    Text(
+                      context.l10n.noConnectionBannerDescription,
+                      style: Theme.of(context).textTheme.bodySmall!.copyWith(
+                            color: VoicesColors.darkTextOnPrimaryLevel0,
+                          ),
+                      softWrap: true,
+                    ),
+                  ],
+                ),
+              ),
+              if (shouldButtonDisplay) const SizedBox(height: 16),
+              if (shouldButtonDisplay)
+                Column(
+                  mainAxisSize: MainAxisSize.min,
+                  children: [
+                    VoicesTextButton(
+                      onTap: onRefresh,
+                      child: Text(
+                        context.l10n.noConnectionBannerRefreshButtonText,
+                        style: Theme.of(context).textTheme.labelLarge!.copyWith(
+                              color: VoicesColors.darkTextOnPrimaryLevel0,
+                            ),
+                      ),
+                    ),
+                  ],
+                ),
```

Comment:
In case you'd like to conditionally include a list of widgets, instead repeating the condition you can just use this pattern, it's either called [destructuring or list spread](https://dart.dev/language/operators#spread-operators):

```suggestion
              if (shouldButtonDisplay) ...[
                 const SizedBox(height: 16),
                 Column(
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    VoicesTextButton(
                      onTap: onRefresh,
                      child: Text(
                        context.l10n.noConnectionBannerRefreshButtonText,
                        style: Theme.of(context).textTheme.labelLarge!.copyWith(
                              color: VoicesColors.darkTextOnPrimaryLevel0,
                            ),
                      ),
                    ),
                  ],
                ),
             ],
```